From e7b3e373d3dcb849089199cbb6fb871c6fba93b4 Mon Sep 17 00:00:00 2001
From: "Joseph K. Bradley" <joseph@databricks.com>
Date: Fri, 15 Apr 2016 11:50:21 -0700
Subject: [PATCH 337/517] [SPARK-14665][ML][PYTHON] Fixed bug with
 StopWordsRemover default stopwords

The default stopwords were a Java object.  They are no longer.

Unit test which failed before the fix

Author: Joseph K. Bradley <joseph@databricks.com>

Closes #12422 from jkbradley/pyspark-stopwords.

(cherry picked from commit d6ae7d4637d23c57c4eeab79d1177216f380ec9c)
Signed-off-by: Joseph K. Bradley <joseph@databricks.com>

Conflicts:
	python/pyspark/ml/feature.py
	python/pyspark/ml/tests.py

(cherry picked from commit 58dfba66ee9b87cdbce6bc2b01324025c7514669)
---
 python/pyspark/ml/feature.py |    2 +-
 python/pyspark/ml/tests.py   |    5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/python/pyspark/ml/feature.py b/python/pyspark/ml/feature.py
index ea874dc..bc63f6d 100644
--- a/python/pyspark/ml/feature.py
+++ b/python/pyspark/ml/feature.py
@@ -1402,7 +1402,7 @@ class StopWordsRemover(JavaTransformer, HasInputCol, HasOutputCol):
         self.caseSensitive = Param(self, "caseSensitive", "whether to do a case " +
                                    "sensitive comparison over the stop words")
         stopWordsObj = _jvm().org.apache.spark.ml.feature.StopWords
-        defaultStopWords = stopWordsObj.English()
+        defaultStopWords = list(stopWordsObj.English())
         self._setDefault(stopWords=defaultStopWords)
         kwargs = self.__init__._input_kwargs
         self.setParams(**kwargs)
diff --git a/python/pyspark/ml/tests.py b/python/pyspark/ml/tests.py
index 674dbe9..b2bd569 100644
--- a/python/pyspark/ml/tests.py
+++ b/python/pyspark/ml/tests.py
@@ -20,6 +20,9 @@ Unit tests for Spark ML Python APIs.
 """
 
 import sys
+if sys.version > '3':
+    basestring = str
+
 try:
     import xmlrunner
 except ImportError:
@@ -283,6 +286,8 @@ class FeatureTests(PySparkTestCase):
         self.assertEqual(stopWordRemover.getInputCol(), "input")
         transformedDF = stopWordRemover.transform(dataset)
         self.assertEqual(transformedDF.head().output, ["panda"])
+        self.assertEqual(type(stopWordRemover.getStopWords()), list)
+        self.assertTrue(isinstance(stopWordRemover.getStopWords()[0], basestring))
         # Custom
         stopwords = ["panda"]
         stopWordRemover.setStopWords(stopwords)
-- 
1.7.9.5

